# Publishes the state of all joints to the /joint_states topic.
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Controller for the main 5 Degrees of Freedom of the arm.
arm_group_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
  gains:
    joint_1: {p: 50000, i: 100, d: 500}
    joint_2: {p: 50000, i: 100, d: 500}
    joint_3: {p: 50000, i: 100, d: 500}
    joint_4: {p: 50000, i: 100, d: 500}
    joint_5: {p: 50000, i: 100, d: 500}

# Controller for the 2-joint gripper.
hand_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - joint_6
    - joint_7
  gains:
    joint_6: {p: 100, i: 0.01, d: 10}
    joint_7: {p: 100, i: 0.01, d: 10}